app/src/main/java/com/artemzin/qualitymatters/ApplicationModule.java

```diff
@@ -33,6 +34,11 @@ public QualityMattersApp provideQualityMattersApp() {
     }
 
     @Provides @NonNull @Singleton
+    public Context provideContext() {
```

Comment:
Well, `Context` is very general class that has too many children: `Activity`, `Service`, etc… I mean, when you receive `context` you can't be sure that it's safe to store the reference to it and not create leaks and so on.

Application object is much more obvious to work with.